### PR TITLE
add net configuration & use tbb for object depth detection algorithm

### PR DESCRIPTION
**Pull request by lspcool**

### CODE REVIEW

File: Sight_++_Solution/Intel_Sight_PP/src/caffe_impl.cpp [hunk 1]
```diff
@@ -42,6 +43,10 @@ struct CaffeModelImpl : public ModelInterface {
 		auto input_blob = cv::dnn::blobFromImage(color_matrix, inScaleFactor, cv::Size(inWidth, inHeight), meanVal, false);
 		net.setInput(input_blob, "data");
 
+		net.setPreferableBackend(cv::dnn::DNN_BACKEND_OPENCV);
```
Comment (GisliGeorgs) [inline Sight_++_Solution/Intel_Sight_PP/src/caffe_impl.cpp:46]: Does this need to be done in each call to the do_work?
Does this not belong in the constructor?

Comment (GisliGeorgs) [reply Sight_++_Solution/Intel_Sight_PP/src/caffe_impl.cpp:46]: and the setPreferableTarget

**Review by GisliGeorgs — APPROVED** [review]

File: Sight_++_Solution/Intel_Sight_PP/src/caffe_impl.cpp [hunk 2]
```diff
@@ -50,68 +54,78 @@ struct CaffeModelImpl : public ModelInterface {
...
+
+
+					// replace pixel values with their center value:
+					float* p = pixel_array.ptr<float>();
+					int group_zero = 0;
+					int group_one = 0;
+					for (size_t i = 0; i < pixel_array.rows; i++) {
+						int center_id = label.at<int>(i);
+						if (center_id == 0) {
+							group_zero++;
```
Comment (GisliGeorgs) [inline Sight_++_Solution/Intel_Sight_PP/src/caffe_impl.cpp:107]: 7 indents is a lot 😃

**Review by GisliGeorgs — APPROVED** [review]

File: Sight_++_Solution/Intel_Sight_PP/src/caffe_impl.cpp (hunk 2 above)
Comment (lspcool) [reply Sight_++_Solution/Intel_Sight_PP/src/caffe_impl.cpp:107]: How can I improve it

Comment (GisliGeorgs) [reply Sight_++_Solution/Intel_Sight_PP/src/caffe_impl.cpp:107]: One is to flip the confidence if statement.
```c++
if(confidence > confidence_threshold) {
  work on confident objects
```
```c++
if(confidence < confidence_threshold) return;
work on confident objects
```
Then you always put things into functions.
But, its okay for now 😄

**Review by GisliGeorgs — APPROVED** [review]
Looking alright.

**Pull request merged**